### ISSUE TITLE
Time plugin cannot be used with Schema-Ast Plugin

### ISSUE THREAD

**Issue by rrrix** [dotansimha/graphql-code-generator#4309]
**Describe the bug**
 
The `time` plugin cannot be used together with the `schema-ast` Plugin. The output of the `time` plugin is hard-coded to begin the line with `//`: 

https://github.com/dotansimha/graphql-code-generator/blob/74b616125f8fe858d5f4f6d9986f80f2514fe2ba/packages/plugins/other/time/src/index.ts#L58
 
**To Reproduce**

```yaml
schema: src/types/*.graphql
document: src/documents/*.graphql
generates:
  ./schema.graphql:
    plugins:
      - time
      - schema-ast
```

The generated schema looks like this (from CodeSandbox example):

https://codesandbox.io/s/silly-haze-j25yb?file=/schema-generated.graphql
```graphql
# The time plugin starts a line with //

// Generated on 2020-06-30T22:32:43+00:00

type Query {
  user(id: ID!): User!
}

type User {
  id: ID!
  username: String!
  email: String!
}
```

3. My `codegen.yml` config file:

https://codesandbox.io/s/silly-haze-j25yb?file=/codegen.yml
```yml
schema: "types/*.graphql"
documents: document.graphql
overwrite: true
generates:
  schema-generated.graphql:
    plugins:
      - add: "# The time plugin starts a line with //"
      - time
      - schema-ast
 ```

**Expected behavior**

Either: 

1. `time` Plugin has an optional prefix argument (default to `//`) that allows user to customize the comment character(s)
2. `time` Plugin auto-detects output format (js, ts, graphql, java, cs, etc.) and uses the appropriate comment format 

**Environment:**

- OS: `MacOS Catalina 10.15.5`
- `@graphql-codegen/...`: `1.16.0`
- NodeJS: `v14.4.0`

**Additional context**

**Comment by dotansimha** [comment]
I guess we need to add support for that. PRs are welcome ;) (You can check the output file extension inside the plugin code, like we are doing in: https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/other/fragment-matcher/src/index.ts#L128)

**Comment by dotansimha** [comment]
Fixed `@graphql-codegen/time@1.17.8`
